### DISCUSSION TITLE
Multiple Selection, pre-populate choices

### DISCUSSION THREAD

**Discussion by mskmkr** [netbox-community/netbox#8643] [Q&A]
I have created a 'Custom Field' named 'Group' with a 'Type' of 'Multiple Selection' with a 'Content Types' of 'Circuits >provider'. 

If I inspect the HTML in chrome, I can see:

![cf_Group](https://user-images.githubusercontent.com/98494314/154078513-a8f811e4-8335-4a7b-bfbf-f7086e9fe420.PNG)

I am trying to populate the choices.

I have the following receiver configured. 

@receiver(models.signals.post_init, sender=Provider)
def my_group(sender, instance, **kwargs):
        GROUP_CHOICES = [grp.name for grp in Group.objects.all()]      
        instance.custom_field_data['Group'] =  SelectMultiple(choices=GROUP_CHOICES)

No choices appear at circuits/providers/add/ 'Custom Fields',' Group*'

What do I need to change in order to pre-populate the choices?

![custom_field_no_options](https://user-images.githubusercontent.com/98494314/154077413-dbdb9254-c4ec-45fe-b24e-93cc8bf3f1de.PNG)

**Comment by candlerb** [comment]
Please can you clarify:
* What is generating this form: a custom script, a plugin, or something else?
* Where are you putting this signal receiver code?
* Where does this "Group" model come from?

> No choices appear at circuits/providers/add/ 'Custom Fields',' Group*'

Modifying the forms of built-in Netbox views is not supported.

As I understand it, Netbox 3.2 will let you create custom fields which reference other database objects. You might need to wait for that.

**Comment by mskmkr** [comment]
1. Plugin
2. plugin/models.py
3. Group.objects.all() is populated from the UI at admin/auth/group/    

GROUP_CHOICES looks like this.
![GROUP_CHOICES](https://user-images.githubusercontent.com/98494314/154088703-e27ec3d4-4df4-46d3-b231-332d3448e691.PNG)
(I have covered providers names)

I have not modified the built-in Netbox form. I added the custom field at /extras/custom-fields/

Hope this helps.
